yogadl/_core.py

```diff
@@ -96,7 +96,7 @@ class Storage(metaclass=abc.ABCMeta):
     in an unspecified format, and returns objects which implement the DataRef
     interface.
 
-    Note that submission and cacheing are not multiprocessing-safe by default.
+    Note that submission and caching are not multiprocessing-safe by default.
     There will be subclasses of Storage which provide such safety, either via
     lock files on a shared filesystem or via some outside worker. In either
```

Comment:
Probably delete this sentence while you are at it.  I had a different architecture in mind when I wrote it than what we actually came up with.

What we ultimately came up with  was that `@cacheable` was always safe and `fetch` and `submit` never were.

Reply:
Tweaked this.